tests/integration/cmd_watch_test.go

```diff
@@ -46,7 +46,19 @@ var _ = Describe("odo watch command tests", func() {
 			helper.CopyExample(filepath.Join("source", "nodejs"), context)
 			helper.CmdShouldPass("odo", "component", "create", "nodejs", "--project", project, "--context", context)
 			output := helper.CmdShouldFail("odo", "watch", "--context", context)
-			Expect(output).To(ContainSubstring("component does not exist. Please use `odo push` to create you component"))
+			Expect(output).To(ContainSubstring("component does not exist. Please use `odo push` to create your component"))
+		})
+	})
+
+	Context("when executing watch without pushing a devfile component", func() {
+		It("should fail", func() {
+			// Devfile push requires experimental mode to be set
+			helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs"), context)
+
+			output := helper.CmdShouldFail("odo", "watch", "--devfile", filepath.Join(context, "devfile.yaml"))
+			Expect(output).To(ContainSubstring("component does not exist. Please use `odo push` to create your component"))
 		})
```

Comment:
@johnmcollier It seems the volume of devfile parser commands are increasing rapidly along with the integration test. So instead of adding the test here in odo regular watch command file which will increase complexity and of course the maintenance cost later on. I have better suggestion 

Let separate out dev file parser in a separate directory. for example 
``` 
$ tree integration/
integration/
[...]
├── debug
│   ├── cmd_debug_test.go
│   └── debug_suite_test.go
├── devfile
│   ├── cmd_devfile_push_test.go
│   └── cmd_devfile_watch_test.go
[...]
```
Also you can add few more basic scenario of devfile watch command like we have done for odo regular watch command in file https://github.com/openshift/odo/blob/master/tests/e2escenarios/e2e_images_test.go#L36. ATM we have only one spec for regular watch command test in watch test file which is not good. Anyway we will add those missing cases shortly may be in the next sprint. Tracking issue is https://github.com/openshift/odo/issues/1819

Reply:
@amitkrout Thanks! I was debating on whether or not to have separate tests for devfile watch, makes sense to me to keep them separate.

I'll create a new stub e2e test file for devfile push too, that includes `odo watch`

Reply:
@amitkrout Actually, I'm wondering if we should hold off on the e2e tests for devfiles for now, until there's a more complete end-to-end scenario? My thinking is that any e2e tests we write right now, wouldn't be reflective of the actual end-to-end scenario for devfiles and odo, and we would need to re-write them once changes like https://github.com/openshift/odo/issues/2734, https://github.com/openshift/odo/issues/2654, https://github.com/openshift/odo/issues/2592 are in.

What I'll do instead is I'll add cases for `odo watch` to the devfile integration test suite for now.

Reply:
agree with @johnmcollier We can keep them here for now.

The e2e tests will need a big overhaul once we have a more complete devfile scenario.
We will probably need a completely new test structure to allow testing on vanilla Kubernetes (see #2752)